### DISCUSSION TITLE
Support for regex POSIX classes. e.g., \p{...}

### DISCUSSION THREAD

**Discussion by messedupryan** [python-openapi/openapi-schema-validator#119] [Q&A] [answered]
I am trying to see if there is any way to support regex POSIX classes like \p{L} or \p{N} with the *open-api-schema-validator*.  In testing, it would appear that the limitation here is that *jsonschema* uses `re` for regex and not the `regex` module.  

Is this something that could be addressed by *open-api-schema-validator*?

For reference, here is an example of what I want to accomplish:

```python
from openapi_schema_validator import validate

schema = {
    "additionalProperties": False,
    "type": "object",
    "properties": {'name': {'description': 'Sample Name',
                        'type': 'string',
                        'pattern': '^([\p{L}\p{Z}\p{N}_.:\/=+\-@]{1,128})$'}
    }
}

doc = {'name': 'some-Name'}

validate(doc, schema)
```

**This throws an exception:**
```
Exception has occurred: SchemaError
'^([\\p{L}\\p{Z}\\p{N}_.:\\/=+\\-@]{1,128})$' is not a 'regex'

Failed validating 'format' in metaschema['allOf'][1]['properties']['properties']['additionalProperties']['$dynamicRef']['allOf'][3]['properties']['pattern']:
    {'format': 'regex', 'type': 'string'}

On schema['properties']['name']['pattern']:
    '^([\\p{L}\\p{Z}\\p{N}_.:\\/=+\\-@]{1,128})$'
re.error: bad escape \p at position 3

The above exception was the direct cause of the following exception:

  File "/Users/terryr/src/openapi-validator-bug/test.py", line 14, in <module>
    validate(claim_xrd, schema)
jsonschema.exceptions.SchemaError: '^([\\p{L}\\p{Z}\\p{N}_.:\\/=+\\-@]{1,128})$' is not a 'regex'

Failed validating 'format' in metaschema['allOf'][1]['properties']['properties']['additionalProperties']['$dynamicRef']['allOf'][3]['properties']['pattern']:
    {'format': 'regex', 'type': 'string'}

On schema['properties']['name']['pattern']:
    '^([\\p{L}\\p{Z}\\p{N}_.:\\/=+\\-@]{1,128})$'
```

The `regex` module is able to handle this.

```python
import regex

print(regex.match('^([\p{L}\p{Z}\p{N}_.:\/=+\-@]{1,128})$', 'foo'))
```

**outputs:**
```
<regex.Match object; span=(0, 3), match='foo'>
```

**Comment by p1c2u** [comment, marked as answer]
Hi @messedupryan 

OpenAPI's regular expressions use [syntax of regex from JavaScript (ECMA 262)](https://json-schema.org/understanding-json-schema/reference/regular_expressions.html).  That means POSIX classes are not part of the standard. You can find equipments in JS [here](https://www.regular-expressions.info/posixbrackets.html). If you require POSIX classes in your case the best option for you would be to extend openapi validator with pattern validator of your choice.
